lib/DirectEditing/PresentationCreator.php

```diff
@@ -0,0 +1,55 @@
+<?php
+/**
+ * @copyright Copyright (c) 2022 Raul Ferreira Fuentes <raul@nextcloud.com>
+ *
+ * @author Raul Ferreira Fuentes <raul@nextcloud.com>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Richdocuments\DirectEditing;
+
+use OCP\DirectEditing\ACreateEmpty;
+use OCP\IL10N;
+
+class PresentationCreator extends ACreateEmpty {
+
+	/**
+	 * @var IL10N
+	 */
+	private $l10n;
+
+	public function __construct(IL10N $l10n) {
+		$this->l10n = $l10n;
+	}
+
+	public function getId(): string {
+		return 'richdocuments_presentation';
+	}
+
+	public function getName(): string {
+		return $this->l10n->t('presentation');
+	}
+
+	public function getExtension(): string {
+		return 'odp';
```

Comment:
Would be good to check for the preferred default document format Open Document Format or Office Open XML and pick the extension / mime type depending on that as done here:

https://github.com/nextcloud/richdocuments/blob/9bb9661795e69d2a035522affe0913626ae3f02c/lib/TemplateManager.php#L459

Reply:
Good call! Added this to the text/presetnation/spreadsheet creators.

Unsure what to do with the graphics one, as far as I know there is no `odg` equivalent in Microsoft Office.